js/vector.js

```diff
@@ -0,0 +1,57 @@
+/*
+    when working w/the vector operations they can
+    accept either one two arguments.
+    if argument.length === 0 then v is expected to be a vector object.
+    if argument.length === 1 then v and y are expected to be numerical values.
+*/
+class Vector {
+
+    constructor(x = 0, y = 0) {
+        this.x = x;
+        this.y = y;
+    }
+
+    add(v, y) {
+        if (arguments.length === 1) {
+            return new Vector(this.x + v.x, this.y + v.y);
+        } else {
+            return new Vector(this.x + y, this.y + y);
+        }
+    }
+
+    subtract(v, y) {
+        if (arguments.length === 1) {
+            return new Vector(this.x - v.x, this.y - v.y);
+        } else {
+            return new Vector(this.x - v, this.y - y);
+        }
+    }
+
+    multiply(scalar) {
+        return new Vector(this.x * scalar, this.y * scalar);
+    }
+
+    divide(scalar) {
+        if (scalar === 0) {
```

Comment:
perhaps add a console log here or something before returning. by surpressing and silencing a potential bug/error, we might shoot ourselves in the foot a bit by not being able to find out where the problem is coming from